tests/src/com/unciv/uniques/GlobalUniquesTests.kt

```diff
@@ -339,4 +341,26 @@ class GlobalUniquesTests {
         Assert.assertTrue(civInfo.gold == 250)
     }
 
+    @Test
+    fun pillageYieldTest() {
+        game.makeHexagonalMap(2)
+        val civInfo = game.addCiv()
+
+        val tile = game.setTileFeatures(Vector2(0f, 0f), Constants.grassland)
+        val cityTile = game.setTileFeatures(Vector2(2f,0f), Constants.grassland)
+        val cityInfo = game.addCity(civInfo, cityTile, true)
+        cityInfo.population.foodStored = 0 // just to be sure
+        civInfo.addGold(-civInfo.gold) // reset gold just to be sure
+
+        val testImprovement = game.createTileImprovement("Pillaging this improvement yields [+20 Gold, +11 Food]")
+        tile.improvement = testImprovement.name
+        val unit = game.addUnit("Warrior", civInfo, tile)
+        unit.currentMovement = 2f
+
+        val pillageAction = UnitActions.getPillageAction(unit)
+        pillageAction?.action?.invoke()
+        Assert.assertTrue(civInfo.gold == 20)
```

Comment:
Could do with a little message? The message could also tell when the getPillageAction went "wrong". Unimportant.

Reply:
Done.